modules/nvidia-gpu-aws-adding-a-gpu-node.adoc

```diff
@@ -0,0 +1,252 @@
+----
+
+. View the machines and machine sets that exist in
+the openshift-machine-api namespace. Each worker machine set is associated with
+a different availability zone within the AWS region. The installer automatically
+load balances workers across availability zones.
++
+[source,terminal]
+----
+$ oc get machinesets -n openshift-machine-api
+----
++
+.Example output
++
+[source,terminal]
+----
+AME DESIRED CURRENT READY AVAILABLE AGE
+
+openshift-blog-txxtf-worker-us-east-2a 1 1 1 1 22m
+
+openshift-blog-txxtf-worker-us-east-2b 1 1 1 1 22m
+
+openshift-blog-txxtf-worker-us-east-2c 1 1 1 1 22m
+----
+
+. At this time, there is only one worker machine per machine set, though a machine set
+could be scaled to add a node in a particular region and zone.
++
+[source,terminal]
+----
+$ oc get machines -n openshift-machine-api | grep worker
+----
++
+.Example output
++
+[source,terminal]
+----
+openshift-blog-txxtf-worker-us-east-2a-8grnb running m4.large us-east-2 us-east-2a 51m
+
+openshift-blog-txxtf-worker-us-east-2b-vw4ph running m4.large us-east-2 us-east-2b 51m
+
+openshift-blog-txxtf-worker-us-east-2c-qpd4q running m4.large us-east-2 us-east-2c 51m
+----
+
+. Make a copy of one of the existing worker
+MachineSet definitions and output the result to a JSON file.
+This will be the basis for the GPU-enabled worker machine set definition.
++
+[source,terminal]
+----
+$ oc get machineset openshift-blog-txxtf-worker-us-east-2a -n openshift-machine-api -o json > <output_file.json>
+----
+
+. Edit the JSON file and make the following changes to the new MachineSet definition:
++
+* Replace ``worker`` with ``gpu``. his will be the name of the new machine set.
+* Change the instance type of the new MachineSet definition to ``p3``,
+which includes an NVIDIA Tesla V100 GPU.
+To learn more about AWS P3 instance types, see
+https://aws.amazon.com/ec2/instance-types/#Accelerated_Computing[Accelerated Computing].
++
+[source,terminal]
+----
+$ jq .spec.template.spec.providerSpec.value.instanceType openshift-blog-txxtf-gpu-us-east-2a.json
+
+"p3.2xlarge"
+----
+
+. Change the name and self link to a unique name that identifies the new
+MachineSet. Delete the status section from the definition.
+Perform a diff of the original worker definition and the new GPU-enabled node definition.
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get machineset openshift-blog-txxtf-worker-us-east-2a -o json | diff openshift-blog-txxtf-gpu-us-east-2a.json -
+----
++
+.Example output
++
+[source,terminal]
+----
+10c10
+
+< "name": "openshift-blog-txxtf-gpu-us-east-2a", --- > "name": "openshift-blog-txxtf-worker-us-east-2a",
+
+13c13
+
+< "selfLink": "/apis/machine.openshift.io/v1beta1/namespaces/openshift-machine-api/machinesets/openshift-blog-txxtf-gpu-us-east-2a", --- > "selfLink": "/apis/machine.openshift.io/v1beta1/namespaces/openshift-machine-api/machinesets/openshift-blog-txxtf-worker-us-east-2a",
+
```

Comment:
remove lines 125-127.  No need to modify the self link, it will be updated when the definition json file is applied.